### DISCUSSION TITLE
Charging data

### DISCUSSION THREAD

**Discussion by TheDailySpank** [tdorssers/TeslaPy#50] [Q&A]
Has anyone been able to get the charging data from the API?

Specifically the CHARGING_HISTORY, CHARGING_HISTORY_VEHICLES, DOWNLOAD_CHARGING_INVOICE, and CHARGING_DOWNLOAD_CSV endpoints?

All I get are 400 errors.

```
vehicles[0].api('CHARGING_DOWNLOAD_CSV')

...

requests.exceptions.HTTPError: 400 Client Error: 400. Oops! Something went wrong. 7bef7898c60d6d35c026e307140b8201 for
url: https://owner-api.teslamotors.com/bff/v2/mobile-app/charging/export
```

**Comment by parna123** [comment]
It looks like CHARGING_HISTORY now works! thanks @tdorssers. Wondering if there is a parameter to get data from earlier months.

**Reply by TheDailySpank** [reply]
CHARGING_HISTORY was added right after I posted this question. :) 

Still looking to get the invoice type info from Tesla as I would like to make a dashboard with the actual costs to me for charging (I've been doing it by hand and it's tedious).

I know they can't tell me how much a charge at a third party costs, but I currently only use free 3rd party chargers, Supercharge or charge at home. I know the $/kWh at home can calculate based on data that is currently available but the Superchargers are all a little different and one is 1/2 price of the rest in my area.

**Reply by parna123** [reply]
+1 I am trying to do something similar. 🙂

Today I am getting the same error back for CHARGING_HISTORY
😞

HTTPError: 400 Client Error: 400. Oops! Something went wrong. 3d626f0ad75e5c7475328881383b796c for url: https://owner-api.teslamotors.com/bff/v2/mobile-app/charging/history

**Reply by hummeling** [reply]
> It looks like CHARGING_HISTORY now works! thanks @tdorssers. Wondering if there is a parameter to get data from earlier months.

Would indeed be great if the API (which I love, btw) could also retrieve charging history for previous months. The data it is in fact available and you can see it in the Tesla app when you swipe the bar graph to the side. Bit of a hidden feature.
